### ISSUE TITLE
Add and hook up the Change groups CTA, integrating the Selection Panel and its core happy path functionality

### ISSUE THREAD

**Issue by techanvil** [google/site-kit-wp#8158]
## Feature Description

Add the **Change groups** CTA to the Audiences Widget Area and open the Selection Panel upon clicking it. Hook up the Selection Panel CTAs for saving and cancelling.

See [audiences widget area > change groups CTA](https://docs.google.com/document/d/1MGD5Djy6AeeZC4zBtHqS-lQEWD9jw0kf-IIWw-jLCFU/edit#heading=h.an9ygf8x05e8) and [selection panel](https://docs.google.com/document/d/1MGD5Djy6AeeZC4zBtHqS-lQEWD9jw0kf-IIWw-jLCFU/edit#heading=h.qt81tly4no4r) in the design doc.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

#### Visibility and Functionality of Change Groups CTA

* The "Change groups" CTA should be visible within the Audiences Widget Area only when there is a non-zero number of available audiences.
* This CTA should not appear if an error occurs while retrieving available audiences.
* Clicking the "Change groups" CTA should open the Selection Panel.
* The design of the CTA should adhere to the [Figma design](https://www.figma.com/file/7pSrkEy8t00BcYRAi9LjjH/Audience-Segmentation?type=design&node-id=1398-26857&mode=design&t=gogn2HjUOVzfaI3X-0.).

#### Selection Panel CTA Functionalities

* **Save Selection CTA**: Upon clicking, this should dispatch the `saveAudienceSettings()` action to persist the local state of the audience settings via the `POST:audience-settings` endpoint. This action will also update the audience tiles to reflect the new selection and close the Selection Panel. The `saveAudienceSettings()` action is being implemented in #8176.
* **Cancel CTA**: Clicking Cancel will close the Selection Panel without making any changes.
* Note that the Selection Panel is being implemented in #8157.

## Implementation Brief

* [ ] Create a new component, `ChangeGroupsLink` in the `assets/js/modules/analytics-4/components/audience-segmentation/dashboard` directory with the following:
    * [ ] Use `assets/js/components/KeyMetrics/ChangeMetricsLink.js` as a starting point.
    * [ ] Use the `getAvailableAudiences` selector of the `analytics-4` store to retrieve the available audiences. This selector is being modified to check if the "Purchasers" audience has had any data in #8157.
    * [ ] Define a new key that will be used to control the open/close state for the selection panel in the `CORE_UI` datastore, say `AUDIENCES_SELECTION_PANEL_OPENED_KEY`.
    * [ ] Display the link only if the following conditions are met:
        * [ ] `analytics-4` is connected
        * [ ] There was no error while obtaining the audiences. Use the `getErrorForSelector` selector of the `analytics-4` datastore, and pass the `getAvailableAudiences` selector with no arguments.
        * [ ] The number of available audiences is greater than 0.
* [ ] Add `assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-segmentation-change-groups-cta.scss`:
    * [ ] Apply the styles to match the [Figma design](https://www.figma.com/file/7pSrkEy8t00BcYRAi9LjjH/Audience-Segmentation?type=design&node-id=1398-26857&mode=design&t=gogn2HjUOVzfaI3X-0.) provided in AC.
* [ ] Import and use the `ChangeGroupsLink` component in the new widget area implemented in [#8138](https://github.com/google/site-kit-wp/issues/8138)
* [ ] Update the Audience Selection Panel's Footer component, which is being implemented in [#8157](https://github.com/google/site-kit-wp/issues/8157) with the following functionalities:
    * [ ] Saving and canceling the audience settings behavior should be similar to `assets/js/components/KeyMetrics/MetricsSelectionPanel/Footer.js.`
    * [ ] **Save Selection CTA**: Dispatch the `saveAudienceSettings()` action of the `analytics-4` store to persist the local state of the audience settings via the `POST:audience-settings` endpoint.
    * [ ] **Cancel CTA**: Close the Audience Selection Panel without making any changes.

### Test Coverage

* Add stories for the new `ChangeGroupsLink` component.
* Add tests for the new `ChangeGroupsLink` component.

## QA Brief

* Set up Site Kit with the Analytics module and turn on the `audienceSegmentation` feature flag.
* In the SK dashboard, scroll down to the audiences widget area and verify that you see a "Change groups" CTA button.
* Verify that clicking on it opens the audience selection panel and includes the audiences that are available with your Analytics property. At a minimum, it should include the `All visitors` audience. It may include the `Purchasers` audience in case there is enough data for it (depending on your property).
* Switch to a property that has all the Site Kit audiences created, or create the Site Kit audiences yourself following the steps [here](https://docs.google.com/document/d/1MGD5Djy6AeeZC4zBtHqS-lQEWD9jw0kf-IIWw-jLCFU/edit#heading=h.e98hcflf4q76). Additionally, create a custom made-up audience.
  * Now, open the audience selection panel again (as instructed above) and verify that the panel includes all the audiences.
  * Verify that the audience source (e.g. created by Site Kit) shows up according to the ACs under each audience.
* If you have an Analytics property with the Site Kit audiences created and data collected for them, please use that as well and verify that the user count shows up beside each audience as applicable.
* Choose different combinations of groups in the selection panel and save your selection. Verify that the selected groups appear in the audiences widget area.

## Changelog entry

* Introduce Audience selection panel in Site Kit dashboard.

**Comment by eugene-manuilov** [comment]
> Selection Panel CTA Functionalities

@hussain-t, I believe the Selection Panel CTA is going to be implemented in another ticket, right? Let's add a reference to that ticket to make it easier to understand where to find it.

**Comment by hussain-t** [comment]
Thanks, @eugene-manuilov. This ticket is already blocked by [#8157](https://github.com/google/site-kit-wp/issues/8157), which is to implement the selection panel. However, I have added a reference to the AC as well.

**Comment by eugene-manuilov** [comment]
Thanks, @hussain-t. AC ✔️

**Comment by techanvil** [comment]
Hi @zutigrm, thanks for drafting this IB.

A minor point - as per the new [code organization](https://docs.google.com/document/d/1MGD5Djy6AeeZC4zBtHqS-lQEWD9jw0kf-IIWw-jLCFU/edit#heading=h.mlj74m7pz9gr) section in the design doc, the SASS file for the component should be in `assets/sass/components/analytics-4/audience-segmentation/`.

More significantly, the determination of the available audiences is not as straightforward as simply retrieving the list from the API. The "Purchasers" default audiences required some special handling. This is discussed in the [available audiences](https://docs.google.com/document/d/1MGD5Djy6AeeZC4zBtHqS-lQEWD9jw0kf-IIWw-jLCFU/edit#heading=h.a4dhnxymldfe) section in the design doc.

However, the implementation for this is not expected to be defined in this issue - rather it will be handled in https://github.com/google/site-kit-wp/issues/8157 which is a dependency of this issue.

I would imagine we'll need to get #8157 and notably its dependency #8141 defined before we can complete the IB for this one.

**Comment by zutigrm** [comment]
@techanvil Thanks, noted. I will hold off until two mentioned issues are defined before updating this one.

**Comment by zutigrm** [comment]
Unassigning myself as this will go to the squad 2. The dependencies mentioned in the last comment are still not defined

**Comment by hussain-t** [comment]
Hi @techanvil. The IB has been updated and is back with you for review.

**Comment by techanvil** [comment]
Thanks @hussain-t, this IB is generally looking good.

One thing though, as per my previous [comment](https://github.com/google/site-kit-wp/issues/8158#issuecomment-1927156487), we need to ensure that the list of available audiences takes into account the requirement for Purchasers to be "active" i.e. that it's had some users since the property was created.

The logic for determining this is still being defined in #8157 and we should be reusing the same logic here. I think we still need to wait until it's fully defined in #8157 before we can comfortably complete the IB for this issue.

**Comment by hussain-t** [comment]
Thanks, @techanvil. The `getAvailableAudiences` selector is being modified for the abovementioned logic in [#8157](https://github.com/google/site-kit-wp/issues/8157). I have added a note to the IB.

**Comment by techanvil** [comment]
Thanks @hussain-t! The IB LGTM. :white_check_mark:

**Comment by nfmohit** [comment]
@techanvil Quick question: Should we update the "Change Metrics" CTA (for KM) as part of this issue? I feel this may create an inconsistency if we don't. I've noticed two differences between the current state and the Figma designs linked here:

1. The "Metrics" is capitalised in the current state, whereas it is lowercase in the Figma designs.
2. The font-family is `Google Sans Display` in the current state, whereas it is lowercase in the Figma designs.

**Comment by sigal-teller** [comment]
@nfmohit The "Metrics" shouldn't be capitalized. In CTA buttons we also capitalized only the first word. As for the font can you share a link to Figma where the font has changed so I can take a look? It should be "label medium" from the design system.

**Comment by nfmohit** [comment]
> @nfmohit The "Metrics" shouldn't be capitalized. In CTA buttons we also capitalized only the first word. As for the font can you share a link to Figma where the font has changed so I can take a look? It should be "label medium" from the design system.

Thanks @sigal-teller! [This](https://www.figma.com/file/7pSrkEy8t00BcYRAi9LjjH/Audience-Segmentation?type=design&node-id=1398-26857&mode=design&t=gogn2HjUOVzfaI3X-0.) is the Figma design I was referring to, specifically the "Change metrics" link. I was actively referring to the font face.

**Comment by sigal-teller** [comment]
Here are 2 screenshots of how it looks for me in Figma. Both CTAs have the same font style and size which is what defined in the [design system](https://www.figma.com/design/ZL8qlbUyR9lmBaF95t0xpq/sitekit-system-gm2%2B?node-id=398-250&t=K6cgRRyRMdSK13dl-1) as "Label/medium" which is defined as Google Sans Text. medium, 14 
<img width="558" alt="Screenshot 2024-06-10 at 17 49 48" src="https://github.com/google/site-kit-wp/assets/69191559/197ee557-fdbc-47c8-bf47-8a51e98589de">
<img width="635" alt="Screenshot 2024-06-10 at 17 49 39" src="https://github.com/google/site-kit-wp/assets/69191559/596744c5-a3f9-487d-ab7e-9f76b442f9ef">

**Comment by kelvinballoo** [comment]
### QA Update ❌ 

I ran a test on this and I have some observations and queries. 
They are documented below

**ITEM 1 - Clarification** ⚠️ 
I am focusing on this point: 
`The "Change groups" CTA should be visible within the Audiences Widget Area only when there is a non-zero number of available audiences.`
What are we implying by 'non-zero'? Does it mean as long as we have at least one audience? Or we are referring to the number of users in an audience between at least one?
________________________________________________________________

**ITEM 2 - Clarification** ⚠️ 
Regarding the point : 
`This CTA should not appear if an error occurs while retrieving available audiences.`
Is there a way to simulate an error on my end for testing this point?
________________________________________________________________

**ITEM 3 - Clarification** ⚠️ 
For this point:
`Save Selection CTA: Upon clicking, this should dispatch the saveAudienceSettings() action to persist the local state of the audience settings via the POST:audience-settings endpoint. `

I did see an audience-settings endpoint request being triggered. It's attached below with status 200.
Is that the request we are to validate?
<Details>
<img width="1176" alt="Screenshot 2024-06-11 at 21 39 11" src="https://github.com/google/site-kit-wp/assets/125576926/b7b3bb73-39ed-4a68-8bea-556c1b2df41a">
</Details>
________________________________________________________________

**ITEM 4 - Clarification** ⚠️ 
My own created audience shows up as 'Already exists in your Analytics property'.
Is that the correct string to be displayed? I did not see it in the design doc.

<Details>
<img width="1166" alt="Screenshot 2024-06-11 at 19 04 15" src="https://github.com/google/site-kit-wp/assets/125576926/3d65e75b-2b28-4892-9b44-a5e394726a3a">
</Details>
________________________________________________________________

**ITEM 5 - Bug** ❌ 
I linked up an analytics property and there are some users when I check it in Analytics.
However, those numbers do not appear in the selection panel.
All the numbers in the selection panel are 0.

<Details>
Analytics shows values for the audiences:
<img width="1156" alt="Screenshot 2024-06-11 at 19 09 05" src="https://github.com/google/site-kit-wp/assets/125576926/88ea4cdc-70ca-4a5b-a312-6d59f90a5ba1">

Selection panel are 0. Please disregard the red selection on the screenshot:
![Uploading Screenshot 2024-06-11 at 19.04.15.png…]()

Tiles are also showing 0:
<img width="1158" alt="Screenshot 2024-06-11 at 19 15 58" src="https://github.com/google/site-kit-wp/assets/125576926/1573805f-201c-471b-a556-e5f2f81c6bfd">

</Details> 

________________________________________________________________

**ITEM 6 - Bug** ❌ 
Here's my scenario:

- I set up an audience tile
- I go to analytics and connect another property
- I go back to my audience tile on the dashboard and the tile is still showing with the title being blank

<Details>
<img width="998" alt="Screenshot 2024-06-11 at 19 13 42" src="https://github.com/google/site-kit-wp/assets/125576926/9716badd-8b12-4941-a2ca-034c7ecdbc2f">

</Details>

________________________________________________________________

**ITEM 7 - Visual Issue** ❌ 
Distance between **New badge** and text should be 14px instead of 6px.
<Details>
Implementation: 6px
<img width="1003" alt="Screenshot 2024-06-11 at 22 05 59" src="https://github.com/google/site-kit-wp/assets/125576926/d356f545-f7fe-4a2b-b76d-8db151a5f404">

Figma: 14px
<img width="598" alt="Screenshot 2024-06-11 at 22 05 10" src="https://github.com/google/site-kit-wp/assets/125576926/5ea5ed78-14dc-46f8-ad66-7d073f73781e">

</Details>

________________________________________________________________

**ITEM 8 - Visual Issue** ❌ 
Margin on top of text should be 32px and not 24px.

<Details>
Implementation : 24px
<img width="1156" alt="Screenshot 2024-06-11 at 18 22 35" src="https://github.com/google/site-kit-wp/assets/125576926/b11cfac4-7e14-45e6-94a4-147d418844fd">

Figma: 32px
<img width="627" alt="Screenshot 2024-06-11 at 22 08 17" src="https://github.com/google/site-kit-wp/assets/125576926/89586001-11f6-4da0-8ef2-fda459f2d725">
</Details>

________________________________________________________________

**ITEM 9 - Visual Issue** ❌ 
Distance between pen icon and 'Change groups' text should be 5px from the pen. There should be no gap.

<Details>
Implementation: notice that there is a gap before the 5px margin:
<img width="299" alt="Screenshot 2024-06-11 at 22 09 41" src="https://github.com/google/site-kit-wp/assets/125576926/b12bc6c3-07a9-4bbe-839a-60632067d8b4">

Figma:
<img width="315" alt="Screenshot 2024-06-11 at 22 11 11" src="https://github.com/google/site-kit-wp/assets/125576926/6a63beaa-bb14-446e-b5b1-a5b12648d3e0">

</Details>

________________________________________________________________

**ITEM 10 - Visual Issue** ❌ 
The 'Change groups' should appear at the bottom of the text on mobile and tablet.
Currently, it appears at the top.

<Details>
Implementation:
<img width="483" alt="Screenshot 2024-06-11 at 19 17 43" src="https://github.com/google/site-kit-wp/assets/125576926/700bc45e-79a8-4212-a454-c5c69a0d9ee6">


Figma:
<img width="210" alt="Screenshot 2024-06-11 at 19 18 16" src="https://github.com/google/site-kit-wp/assets/125576926/791670f6-2e20-4f7b-8511-51760cc28898">

</Details>

**Comment by nfmohit** [comment]
Hi @kelvinballoo, thank you for sharing your observations.

> **ITEM 1 - Clarification** ⚠️
> I am focusing on this point:
> `The "Change groups" CTA should be visible within the Audiences Widget Area only when there is a non-zero number of available audiences.`
> What are we implying by 'non-zero'? Does it mean as long as we have at least one audience? Or we are referring to the number of users in an audience between at least one?

It refers to the number of available audiences/groups, not the number of users.

> **ITEM 2 - Clarification** ⚠️
> Regarding the point :
> `This CTA should not appear if an error occurs while retrieving available audiences.`
> Is there a way to simulate an error on my end for testing this point?

It's slightly complicated to simulate an error, but you can simulate a state where there are no available audiences (which will basically result due to an error). To do so, run the following script in the browser console:

```js
googlesitekit.data.dispatch( 'modules/analytics-4' ).setSettings( {
    availableAudiences: null,
} );
```

You should see that the CTA has disappeared.

> **ITEM 3 - Clarification** ⚠️ For this point: `Save Selection CTA: Upon clicking, this should dispatch the saveAudienceSettings() action to persist the local state of the audience settings via the POST:audience-settings endpoint. `
> 
> I did see an audience-settings endpoint request being triggered. It's attached below with status 200. Is that the request we are to validate?

That is correct.

> **ITEM 4 - Clarification** ⚠️ My own created audience shows up as 'Already exists in your Analytics property'. Is that the correct string to be displayed? I did not see it in the design doc.

That is the correct string according to the Figma designs.

> **ITEM 5 - Bug** ❌
> I linked up an analytics property and there are some users when I check it in Analytics.
> However, those numbers do not appear in the selection panel.
> All the numbers in the selection panel are 0.

Analytics information that is available in analytics.google.com does not reflect right away in Site Kit, as there is a 24-48 hour delay for it to be available in the reporting API. If the respective metrics are also zero in other parts of Site Kit (e.g. Audience tiles, according to your third screenshot in this section), it is expected for them to also be zero in the selection panel.

> **ITEM 6 - Bug** ❌ Here's my scenario:
> 
> * I set up an audience tile
> * I go to analytics and connect another property
> * I go back to my audience tile on the dashboard and the tile is still showing with the title being blank

This is indeed a bug, but out of the scope of this issue. Ideally, when an Analytics property is changed in Site Kit, the list of configured audiences should also be reset. Could you open a new issue regarding this? Thanks!

> **ITEM 7 - Visual Issue** ❌
> Distance between **New badge** and text should be 14px instead of 6px.

This is also out of the scope of this issue, as this issue is only concerned about the "Change groups" CTA button and the selection panel that it opens. However, I believe this uses the standard new badge placement procedure similar to any other widget which was likely set to 6 pixels before. If we had to change it, we'd have to change it in the widget area registration function so that it changes for any other widget area as well. You can open a separate issue for this as well.

> **ITEM 8 - Visual Issue** ❌
> Margin on top of text should be 32px and not 24px.

This is also out of the scope of this issue. You may open a separate issue for this.

> **ITEM 9 - Visual Issue** ❌
> Distance between pen icon and 'Change groups' text should be 5px from the pen. There should be no gap.

According to Figma (I'm using "Dev mode"), I can see that there's a 6 pixel gap. Screenshot:

![image](https://github.com/google/site-kit-wp/assets/20284937/ac796876-db9e-48a1-a8e3-f90229e13a33)

The icon, when used during implementation, has a spacing of around 1 pixel around the subject to account for non-square graphics. Also, this is the same button that we use for the "Change metrics" CTA in the Key Metrics section if you scroll a little above in your Site Kit dashboard (with KM configured). If we had to change anything here, it'd lose consistency. In short, I believe this is fine.

> **ITEM 10 - Visual Issue** ❌
> The 'Change groups' should appear at the bottom of the text on mobile and tablet.
> Currently, it appears at the top.

This is something that I think needs a little confirmation from @sigal-teller and @techanvil. The CTA button here is attached to the widget area. We can make it show up in the footer of the widget area for mobile breakpoints (this will also apply globally and will impact the KM "Change metrics" CTA as well), but the Audience Segmentation widget area also has an additional `InfoNoticeWidget` underneath the audience tiles, so the "Change groups" CTA will appear below the `InfoNoticeWidget`. Is it okay to show it there, or do we want to do something else here?

Here's a mock of what it may look:
![image](https://github.com/google/site-kit-wp/assets/20284937/9595b36d-ad05-41fc-b473-4d6c81eb79fd)

**Comment by nfmohit** [comment]
Hi @techanvil and @sigal-teller.

I have co-assigned you both to this issue regarding a difference between the Figma designs and implementation, as follows:

> **ITEM 10 - Visual Issue** ❌ The 'Change groups' should appear at the bottom of the text on mobile and tablet. Currently, it appears at the top.
> 
> Implementation: <img alt="Screenshot 2024-06-11 at 19 17 43" width="483" src="https://private-user-images.githubusercontent.com/125576926/338699374-700bc45e-79a8-4212-a454-c5c69a0d9ee6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxMzAzNzYsIm5iZiI6MTcxODEzMDA3NiwicGF0aCI6Ii8xMjU1NzY5MjYvMzM4Njk5Mzc0LTcwMGJjNDVlLTc5YTgtNDIxMi1hNDU0LWM1YzY5YTBkOWVlNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYxMVQxODIxMTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05OTFiMWNiMWUwZmRkMzI4YjA0YzZkMDQzNzlhYzA3ZjA2NTQyYjFjNGU0ZWNhMzI2NDZlNTI2MjJmYTg1NmI0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.NuvKcA_3bPN7n9kkky9FilvoS1TEat_VTBFoOz22GNk">
> Figma: <img alt="Screenshot 2024-06-11 at 19 18 16" width="210" src="https://private-user-images.githubusercontent.com/125576926/338699044-791670f6-2e20-4f7b-8511-51760cc28898.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxMzAzNzYsIm5iZiI6MTcxODEzMDA3NiwicGF0aCI6Ii8xMjU1NzY5MjYvMzM4Njk5MDQ0LTc5MTY3MGY2LTJlMjAtNGY3Yi04NTExLTUxNzYwY2MyODg5OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYxMVQxODIxMTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xOTQzY2RiNGZlYzg3MTBjNzMwOTAwYWE4MDk5MDlmMjczNTBhM2FjOTI3N2NmNjgwMmEzYzZiMDJmMTAwMDFmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.j5As3C4Cci0EFhtFNqZsdhkfuJwwbGQ4SEIlvBVxU64">

The CTA button here is attached to the widget area. We can make it show up in the footer of the widget area for mobile breakpoints (this will also apply globally and will impact the KM "Change metrics" CTA as well), but the Audience Segmentation widget area also has an additional InfoNoticeWidget underneath the audience tiles, so the "Change groups" CTA will appear below the InfoNoticeWidget. Is it okay to show it there, or do we want to do something else here?

Thank you!

**Comment by kelvinballoo** [comment]
**ITEM 1 - ✅** 
Noted. I have tried to test a scenario with 0 audience but it seems that this is not possible as there will always be a defaulted 'All Users' audience. That is fine.
__________________________________________

**ITEM 2 - ✅** 
Yes, indeed after executing the null script, the CTA would disappear.
```
googlesitekit.data.dispatch( 'modules/analytics-4' ).setSettings( {
    availableAudiences: null,
} );
```

<Details>
CTA disappears after executing script.
<img width="1165" alt="Screenshot 2024-06-12 at 15 21 07" src="https://github.com/google/site-kit-wp/assets/125576926/92922655-92cd-4d8c-85df-9a61013b94f8">

</Details>

__________________________________________

**ITEM 3 - ✅** 
Noted. No action here.

__________________________________________

**ITEM 4 - ✅**
Got it. I looked in the wrong window with the many designs. Found out the correct design and it's as expected.
<Details>
Figma wording is same as implemeted : "Already exists in your Analytics property"
<img width="335" alt="Screenshot 2024-06-12 at 15 24 39" src="https://github.com/google/site-kit-wp/assets/125576926/e5a4a643-9524-401b-b780-a78f4e4dc998">

</Details> 

__________________________________________

**ITEM 5 - ⚠️ **
I want to drill down a bit more on this as it's been already 24 hours since I spun up the site and in the last 28 days, there's been 3 users for 'All users' and 'New Visitors'.
However, these are still not reflected in the selection panel, even if I set up my SK dashboard to last 28 days too.

Would you rather I wait another 24 hours @nfmohit to confirm if this is working or not? Or is this out of scope for this release?
<Details>
3 users shown on GA:
<img width="1106" alt="Screenshot 2024-06-12 at 15 30 46" src="https://github.com/google/site-kit-wp/assets/125576926/c2baedaa-978f-407e-af2c-35e8b6fd1f3f">

0 shown on selection panel:
<img width="1154" alt="Screenshot 2024-06-12 at 15 30 58" src="https://github.com/google/site-kit-wp/assets/125576926/909f73ba-8a01-464f-9afa-f92ee65bd8af">

</Details>

__________________________________________

**ITEM 6 - ✅  **
Issue raised here: https://github.com/google/site-kit-wp/issues/8860

__________________________________________

**ITEM 7 - ✅  **
Issue raised here: https://github.com/google/site-kit-wp/issues/8861

__________________________________________

**ITEM 8 - ✅  **
Issue raised here: https://github.com/google/site-kit-wp/issues/8862
__________________________________________

**ITEM 9 - ✅   **
Just to clarify what I was trying to say was that on the implementation, we have possibly a gap of 10px roughly from the pen to the CTA.
On Figma, it's around 5px (if 6px from dev figma, that's fine) but there is this extra gap denoted in blue below. The orange section is 5px. The blue arrow is roughly around that size too, totalling 10px. (Image below)

That said, noted taken on what you mentioned about consistency when comparing with KM selection. The figma design is not accurate here.

<Details>
<img width="284" alt="Screenshot 2024-06-11 at 22 09 41" src="https://github.com/google/site-kit-wp/assets/125576926/a60e05a9-ff04-409c-8888-f0773bd54b2c">

KM section: There is the gap as well:
<img width="437" alt="Screenshot 2024-06-12 at 16 00 12" src="https://github.com/google/site-kit-wp/assets/125576926/a30afaa9-cc1a-4d7f-9375-b03c1d0b08a8">

</Details>

__________________________________________

**ITEM 10 - ⚠️    **
Issue raised under : https://github.com/google/site-kit-wp/issues/8863
@nfmohit , I did not paste the comment you had over there because I did not understand everything. 
Also, not sure exactly whether to post in comments or somewhere in the ticket. So maybe you can follow up there with your comment.

**Comment by kelvinballoo** [comment]
### QA Update ✅ 

All the items have been acted upon now.

Here's an update on the pending items in the earlier comment:
**ITEM 5** ✅ 
This is likely a setup issue. After using another property, the values were pulled on the selection panel and tiles.

<Details>
<img width="969" alt="Screenshot 2024-06-13 at 10 29 32" src="https://github.com/google/site-kit-wp/assets/125576926/595a8a38-ffee-41fc-a88a-3035678013ac">

<img width="494" alt="Screenshot 2024-06-13 at 10 44 18" src="https://github.com/google/site-kit-wp/assets/125576926/7408062c-2be7-4c30-952f-6506a10a5ce2">
</Details>

**ITEM 10** ✅ 
Comment has been updated in https://github.com/google/site-kit-wp/issues/8863 by Nahid

_____________________________________________________
As for the points under the QAB, the following were already met accordingly:
- The feature is working well with the audienceSegmentation feature flag on and CTA disappears when the flag is off. 
- The "Change groups" CTA button is presenet.
- Clicking on it opens the audience selection panel and includes the audiences that are available with your Analytics property. At a minimum, it should include the All visitors audience.
- Custom audiences also appear based on what is in Analytics

Moving this ticket to Approval.

<Details>
Change groups CTA is present:
<img width="1174" alt="Screenshot 2024-06-11 at 18 04 02" src="https://github.com/google/site-kit-wp/assets/125576926/1747df37-37d4-49a7-8ec1-27154118a06a">

Data is shown in the panel and on the tiles:
<img width="494" alt="Screenshot 2024-06-13 at 10 44 18" src="https://github.com/google/site-kit-wp/assets/125576926/9fb39ba5-c1a7-4b78-87ca-9d8ac8502a81">

<img width="969" alt="Screenshot 2024-06-13 at 10 29 32" src="https://github.com/google/site-kit-wp/assets/125576926/2cd1e020-039d-4325-90d9-cc12a80df922">

Custom audiences appear as well:
<img width="1161" alt="Screenshot 2024-06-11 at 19 03 04" src="https://github.com/google/site-kit-wp/assets/125576926/d2e6d60c-c39f-47e7-adb0-ce64abeb4fd2">

</Details>
